includes/class-access-token.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * File containing the class Access_Token.
+ *
+ * @package wp-job-manager
+ */
+
+namespace WP_Job_Manager;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * A token that provides access to a private resource.
+ *
+ * @since $$next-version$$
+ */
+class Access_Token {
+	/**
+	 * Token's metadata. They are hashed together with the token.
+	 *
+	 * @var array
+	 */
+	private array $metadata;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param array $metadata Metadata to be hashed together with the token.
+	 */
+	public function __construct( array $metadata ) {
+		$this->metadata = $metadata;
+	}
+
+	/**
+	 * Creates a new token.
+	 *
+	 * @return string The token.
+	 */
+	public function create() : string {
+		$tick = wp_nonce_tick();
+
+		return $this->generate_token( $tick );
+	}
+
+	/**
+	 * Generates a token for a tick.
+	 *
+	 * @param float $tick
+	 *
+	 * @return false|string
+	 */
+	private function generate_token( float $tick ) {
+		$metadata_json = wp_json_encode( $this->metadata );
```

Comment:
This will create a different output for `[ 'a' => 1, 'b' => 2 ]` and `[ 'b' => 2, 'a' => 1 ]`, which is probably not what you want. Maybe consider using `ksort` before encoding?